app/frontend/entrypoints/plates_from_tubes.js

```diff
@@ -0,0 +1,94 @@
+"use strict";
+
+import $ from "jquery";
+import CodeMirror from "codemirror";
+
+/**
+ * Defines a custom CodeMirror mode called "barcode_reader".
+ * This mode is used to highlight duplicate barcodes in the input.
+ */
+CodeMirror.defineMode("barcode_reader", function (_) {
+  /**
+   * Processes the input stream and determines the style for each token.
+   *
+   * @param {Object} stream - The CodeMirror stream object for reading input.
+   * @param {Object} state - The state object that tracks barcodes.
+   * @returns {string|null} - The style to apply to the token, or null if no style.
+   */
+  function tokenBase(stream, state) {
+    let ch = stream.next(); // Read the next character
+    if (/\w/.test(ch)) {
+      // Check if the character is alphanumeric
+      stream.eatWhile(/[\w.]/); // Continue reading alphanumeric characters
```

Comment:
I think we want to include more than just alphanumeric characters. For instance, dashes are a valid part of barcodes. Perhaps the ones scanned in will just be alphanumeric, but the format could change to include other things in the future e.g. dashes like in the plate barcodes.

<img width="166" alt="Screenshot 2025-04-04 at 09 13 16" src="https://github.com/user-attachments/assets/0d4bd6b0-98e3-4267-ab29-ea65e0df4482" />

What about accepting any character apart from a space or a newline?